### ISSUE TITLE
Image downloading issues

### ISSUE THREAD

**Issue by azurenekowo** [4pii4/piehtvn#3]
Accessing them directly returns a CloudFlare blocked message:
"Sorry, you have been blocked. You are unable to access hentaivn.autos"

![image](https://github.com/4pii4/piehtvn/assets/67498765/59778163-3545-4fa5-a0b0-5d834a682369)

Seems to happen with all chapter URLs I have tested.
`http://127.0.0.1:7479/get-images?url=35804-66596-xem-truyen-sau-khi-co-duoc-kha-nang-tang-hinh-toi-co-the-lam-bat-cu-viec-gi-voi-mot-ban-nu-xinh-dep-nhat-truong-oneshot`
```json
[
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926195-134.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926197-picsart_23-12-17_19-49-02-487.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926197-picsart_23-12-17_20-07-04-642.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926198-picsart_23-12-17_20-29-45-221.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926199-picsart_23-12-18_09-28-31-577.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926200-picsart_23-12-18_09-35-53-009.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926201-picsart_23-12-18_09-47-52-462.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926202-picsart_23-12-18_10-49-56-714.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926203-picsart_23-12-18_10-32-56-313.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926203-picsart_23-12-18_11-05-39-789.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926204-picsart_23-12-19_00-21-16-067.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926205-picsart_23-12-19_00-37-10-398.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926206-picsart_23-12-19_00-37-58-920.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926207-picsart_23-12-19_00-42-49-477.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926207-picsart_23-12-19_00-48-32-392.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926208-picsart_23-12-19_00-54-34-076.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926209-picsart_23-12-19_01-01-07-151.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926210-picsart_23-12-19_01-07-17-553.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926210-picsart_23-12-19_01-28-44-522.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926211-picsart_23-12-19_01-11-24-295.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926212-picsart_23-12-19_01-15-03-708.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926213-picsart_23-12-19_01-20-40-590.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926214-picsart_23-12-19_01-26-38-309.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926214-picsart_23-12-19_01-34-43-679.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926215-picsart_23-12-19_01-40-58-643.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926216-picsart_23-12-19_01-43-48-640.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926217-picsart_23-12-19_01-51-12-664.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926218-picsart_23-12-19_01-51-28-719.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926218-picsart_23-12-19_01-51-46-005.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926219-picsart_23-12-19_01-52-02-565.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926219-picsart_23-12-19_01-55-16-510.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/18/1702926220-picsart_23-12-19_01-55-30-406.jpg",
	"https://ULGREEDFoWs.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/19/1702956029-1701622312-1695534800-1694533850-khong_co_tieu_e1789.png"
]
```

**Comment by 4pii4** [comment]
Look at [Image class](https://github.com/4pii4/piehtvn/blob/master/piehtvn.py#L96) or inspect the image requests in the official reader for those headers.

**Comment by azurenekowo** [comment]
![image](https://github.com/4pii4/piehtvn/assets/67498765/a9086d71-f218-4ae2-a677-040e21d9ac77)

@4pii4 Am I missing something here...?

**Comment by 4pii4** [comment]
@azurenekowo you are missing the `Referrer` header which actually is the most important one. An `Image`'s header is just it's base url though.

<details>
<summary>sample headers</summary>

```
https://evvdsfgefdszihfdx.hentaivn.autos/HGHrrFSDweqZSfd/lgbdfexijzbxsdrl/2023/07v2/30/1690735499-creditkngcfan.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/UGHrtgcrtf/lgbdfexijzbxsdrl/2023/08v2/12/1691846778-1.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/ULGREFDFoVs/lgbdfexijzbxsdrl/2023/07v2/30/1690735502-dchthu.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/IIJFGCFWDFqZav/lgbdfexijzbxsdrl/2023/07v2/30/1690735503-penguincredit2.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/NZu7ULbbqwoV/lgbdfexijzbxsdrl/2023/08v2/13/1691895968-2.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/NZu7ULbbqwoV/lgbdfexijzbxsdrl/2023/08v2/13/1691895970-3.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/ULGREFDFoVs/lgbdfexijzbxsdrl/2023/08v2/13/1691895970-4.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/QQsaxaXKjbj/lgbdfexijzbxsdrl/2023/08v2/13/1691895971-5.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/UGHrtgcrtf/lgbdfexijzbxsdrl/2023/08v2/13/1691895971-6.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/QQsaxaXKjbj/lgbdfexijzbxsdrl/2023/08v2/13/1691895971-7.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/IIJFGCFWDFqZav/lgbdfexijzbxsdrl/2023/08v2/13/1691895972-8.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/UGHrtgcrtf/lgbdfexijzbxsdrl/2023/08v2/13/1691895972-9.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/NZu7ULbbqwoV/lgbdfexijzbxsdrl/2023/08v2/13/1691895973-10.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/HGHrrFSDweqZSfd/lgbdfexijzbxsdrl/2023/08v2/13/1691895973-11.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/EWDSfgeFDSZIHFdx/lgbdfexijzbxsdrl/2023/08v2/13/1691895973-12.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/HGHrrFSDweqZSfd/lgbdfexijzbxsdrl/2023/08v2/13/1691895974-13.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/OOrdOOFGTRFwdeSX/lgbdfexijzbxsdrl/2023/08v2/13/1691895974-14.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/UGHrtgcrtf/lgbdfexijzbxsdrl/2023/08v2/13/1691895975-15.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/QQsaxaXKjbj/lgbdfexijzbxsdrl/2023/08v2/13/1691895975-16.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/PPPOHUerwrefFEwdr/lgbdfexijzbxsdrl/2023/08v2/13/1691895976-17.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/IIJFGCFWDFqZav/lgbdfexijzbxsdrl/2023/08v2/13/1691895976-18.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/UGHrtgcrtf/lgbdfexijzbxsdrl/2023/08v2/13/1691895976-19.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/NZu7ULbbqwoV/lgbdfexijzbxsdrl/2023/08v2/12/1691846792-20.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/UGHrtgcrtf/lgbdfexijzbxsdrl/2023/08v2/12/1691846793-21.jpg {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
https://evvdsfgefdszihfdx.hentaivn.autos/OOrdOOFGTRFwdeSX/lgbdfexijzbxsdrl/2023/07v2/30/1690735529-donate.png {'Accept-Encoding': 'gzip, deflate, br', 'Referer': 'https://evvdsfgefdszihfdx.hentaivn.autos/', 'Sec-Fetch-Dest': 'image', 'Sec-Fetch-Mode': 'no-cors', 'Sec-Fetch-Site': 'same-site', 'Sec-GPC': '1', 'Pragma': 'no-cache', 'Cache-Control': 'no-cache', 'TE': 'trailers', 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0', 'Accept-Language': 'en-US,en;q=0.5', 'Connection': 'keep-alive'}
```

</details>

**Comment by azurenekowo** [comment]
@4pii4 hi, after modifying the `Referer` header to the hostname of the image URL (`xxxxxx.hentaivn.autos`) it produced the same result. The headers are the exact same as the sample you provided above, except for the image URL.
![image](https://github.com/4pii4/piehtvn/assets/67498765/549f79fc-470e-49dd-a069-ba5d2e49821d)

By the way, it seems that those image URLs on your side looks a bit different than mine. (?)
Your URL is `https://xxxxxx.hentaivn.autos/xxxxxx/xxxxxx/2023/08v2/12/1691846793-21.jpg`
Mine (on every URL I had tried) is `https://xxxxxx.hentaivn.autos/i&m+a-g+e!s+1200/2023/12/19/1703013278-pic01.jpg`

**Comment by 4pii4** [comment]
I actually have no idea why it's throwing 521 as I've never encountered that before. How about trying another image?

**Comment by azurenekowo** [comment]
@4pii4 Apparently those URLs are persistent, when requested by my browser (?) The only way to change them is clearing your browser data (cache / cookies) of that site.
I tried but it changed only a few characters (`https://ulgrffdfovvs.hentaivn.autos/` => `https://ulgreedfoows.hentaivn.autos/`).

The omnious `/i&m+a-g+e!s+1200/` remained.
This happens both in Insomnia and my browser. 

## **Update while I was typing out this issue**
The URL suddenly changed to the same as your format. I'm not sure why, given my browser still had those previous URLs.

## **Update 2**
It differs for some doujins?? I think it's because of how their uploading system works. They have multiple servers iirc.

I have tried about 5 different doujins.

`35801-66589-xem-truyen-tan-tinh-em-di-chap-41-end`
![image](https://github.com/4pii4/piehtvn/assets/67498765/bbca3f7c-e5f1-46cb-9426-58a1bab9af61)

`35258-66604-xem-truyen-ngay-may-man-meo-den-2`
![image](https://github.com/4pii4/piehtvn/assets/67498765/2765243c-31bc-44fc-a805-7578fd1d8e00)

`35335-65307-xem-truyen-siblings-switch-oneshot`
![image](https://github.com/4pii4/piehtvn/assets/67498765/4d34977e-283f-4c3d-b1de-56faedb9dce6)

`35528-65641-xem-truyen-nu-anh-hung-cua-em-oneshot`
![image](https://github.com/4pii4/piehtvn/assets/67498765/14f2e76e-c1d6-4e7f-b5a9-493f2d4c8a97)

`33837-64689-xem-truyen-netorimura-chap-4ket-thuc`
![image](https://github.com/4pii4/piehtvn/assets/67498765/965deb36-d924-4c32-b7c8-6855e7ff85a7)

**Comment by 4pii4** [comment]
I don't really know what's wrong with your setup but my test download script is working fine with all of those doujins

![image](https://github.com/4pii4/piehtvn/assets/138131197/6f9cdebd-e7cf-452b-8eac-8bfdae122507)

![image](https://github.com/4pii4/piehtvn/assets/138131197/0d582b04-57ad-4630-b512-19f518fcea27)

![image](https://github.com/4pii4/piehtvn/assets/138131197/c425fd71-631c-4db0-81ba-c00368edbccd)

![image](https://github.com/4pii4/piehtvn/assets/138131197/e2c51877-e36b-4623-b6ac-b1cfdb003de7)

![image](https://github.com/4pii4/piehtvn/assets/138131197/316347b5-58e0-42c1-9499-d9a90e162a77)

**Comment by azurenekowo** [comment]
@4pii4 hmm, can you give me your script?

**Comment by 4pii4** [comment]
@azurenekowo sure, here you go

```py
import requests

from piehtvn import Chapter, DOMAIN, Image

s = requests.Session()

for img in Chapter(None, '33837-64689-xem-truyen-netorimura-chap-4ket-thuc.html', None, DOMAIN).get_images():
    req = Image(img).get_request().prepare()
    res = s.send(req)
    print(img, len(res.content))
```

Put it in the same directory as the other `.py` files

**Comment by azurenekowo** [comment]
@4pii4 apparently appending `.html` at the end solves it.

*(no response)*
![image](https://github.com/4pii4/piehtvn/assets/67498765/929bb83c-7175-42fb-b44f-aafc8e192e2d)

*(appended .html at the end)*
![image](https://github.com/4pii4/piehtvn/assets/67498765/8b0f11e5-8cc5-4568-9289-61d3754aa7c2)

Now I need to find a way to fetch that image in the backend and return it to the frontend (Frontend JS just cries about CORS)

**Comment by 4pii4** [comment]
Well yes, you need .html when using the classes directly but iirc the backend endpoints add .html for you

**Comment by azurenekowo** [comment]
@4pii4 can you try accessing the image with a REST client / download it?

I had shrugged it off as some shenanigans with Insomnia, can you send a request to get the image and save it to your disk?

**Comment by hUwUtao** [comment]
quoted from @azurenekowo 
> @4pii4 can you try accessing the image with a REST client / download it?
> 
> I had shrugged it off as some shenanigans with Insomnia, can you send a request to get the image and save it to your disk?

I did the thing on blogtruyen before. The way: you copy the whole download query (devtool generated) and download as curl, then imitate in fetch (shouldn't because you may get into issues).

So well, I can say I can just do this in a single varnish configuration but so, ima build one.

TLDR: a cache middleware (varnish like so) as-a-proxy

**Comment by hUwUtao** [comment]
ima implement this fast-as-f with rust ;)

**Comment by hUwUtao** [comment]
> ima implement this fast-as-f with rust ;)

https://github.com/huwutao/mangate WIP

**Comment by 4pii4** [comment]
@azurenekowo I have implemented a `/download-image` endpoint like this screenshot so no front end workaround is necessary anymore.
![image](https://github.com/4pii4/piehtvn/assets/138131197/ab343e6b-1005-45c2-9324-68c31671b523)

**Comment by hUwUtao** [comment]
> https://github.com/huwutao/mangate WIP
rebump this, actually a working thing now, but really archival
quick ref: `/api/fetch?https://[the domain].red/[reader-view].html`
it return `{cur: str, ls: str[]}`, request `/static/[cur]/str[n]` to get the image.

note: hvn itself not aware of image formats, like apng/gif might work, and it should. some images cannot be view from normal image viewer, but works on [modern] browser

**Comment by 4pii4** [comment]
I'll code a fallback image downloader similar to the `Server 1/2` button on the official site asap

**Comment by 4pii4** [comment]
issue fixed as of https://github.com/4pii4/piehtvn/commit/39531ba4f0eebbc26c13ff20c9d31f7718558c6e
